submit-api/src/SubmitAPI/Internal/Transaction.hs

```diff
@@ -0,0 +1,190 @@
+module SubmitAPI.Internal.Transaction where
+
+import           RIO
+import qualified Data.Text                 as T
+import qualified Data.Map.Strict           as Map
+import           Codec.Serialise
+import           Data.ByteString.Lazy      (toStrict)
+import           Data.Text.Prettyprint.Doc (Pretty(..))
+import qualified Data.Set                  as Set
+
+import           Cardano.Api                    hiding (TxBodyError)
+import           Cardano.Api.Shelley            (ProtocolParameters)
+import           Cardano.Slotting.Time          (SystemStart)
+import qualified Ledger                         as P
+import qualified Plutus.Contract.CardanoAPI     as Interop
+import           Plutus.Contract.CardanoAPITemp as Interop
+import qualified Ledger.Ada                     as Ada
+
+import qualified CardanoTx.Models   as Sdk
+import           CardanoTx.ToPlutus
+import           NetworkAPI.Env
+
+signTx
+  :: TxBody AlonzoEra
+  -> [ShelleyWitnessSigningKey]
+  -> Tx AlonzoEra
+signTx body keys =
+  makeSignedTransaction wits body
+    where wits = keys <&> makeShelleyKeyWitness body
+
+buildBalancedTx
+  :: MonadThrow f
+  => SystemEnv
+  -> Sdk.ChangeAddress
+  -> Set.Set Sdk.FullCollateralTxIn
+  -> Sdk.TxCandidate
+  -> f (BalancedTxBody AlonzoEra)
+buildBalancedTx SystemEnv{..} defaultChangeAddr collateral txc@(Sdk.TxCandidate{..}) = do
+  let era          = AlonzoEra
```

Comment:
Is it necessary to "harcode" this params?

Reply:
Yep, because SDK depends on Plutus, so other eras are nonsensical